src/mlpack/bindings/python/setup.py.in

```diff
@@ -108,5 +108,7 @@ setup(name='mlpack',
       cmdclass={ 'build_ext': build_ext },
       ext_modules = modules,
       setup_requires=['cython', 'pytest-runner'],
-      tests_require=['pytest'],
+      tests_require=['pytest>4.6;python_version>"2.7"', 'pytest<=4.6;python_version<="2.7"',
```

Comment:
I'm wondering if the ``pytest`` version part is enough to resolve the issue, or if ``more-itertools > ...`` is necessary as well.

Reply:
Hi @zoq I think so when we run Ci-travis for python 2.7 it was downloading more-itertools 7.0.0 that is incompatible with python 2.7.

Reply:
This is more of a general question - Is it important to provide clarity as to which versions are breaking compatibility? In that case we could make it pytest>4.6;python_version>="3.5" since pytest 5.0.0 is only compatible with python 3.5+. It would work either way though.

Reply:
Yeah @favre49  I was thinking for so the same. I will update this Thank You.